### ISSUE TITLE
`VPN` subscription can't be restored if `Leo` subscription is restored first

### ISSUE THREAD

**Issue by hffvld** [brave/brave-browser#37161]
## Description 

I tested multiple scenarios where I purchased a <ins>real</ins> `Leo` and `VPN` subscriptions on the `Web` and then tried to restore it on Android :
- `Refresh Leo` credentials > Restored successfully
- `Restore Brave VPN` credentials > Restored successfully
- Restore both `Leo` and `VPN` credentials > If `VPN` credentials are restored first, then both subscriptions are restored successfully
- Restore both `Leo` and `VPN` credentials > If `Leo` credentials are restored first, then `VPN` is not restored/not working 

## Steps to reproduce 

   1. Have `Leo` and `VPN` subscriptions purchased on the `Web` already
   2. Fresh install and launch Brave
   3. Go to `account.brave.com` > Sign in to your account
   4. Tap `Refresh Leo` > Wait for `You have active credentials loaded!` message
   5. Three-dot menu > Settings > Leo > *Confirm*
   6. Go back to `account.brave.com` > Tap `Refresh Brave VPN` > Wait for `You have active credentials loaded!` message
   7. Three-dot menu > Settings > Brave Firewall + VPN > *Observe that subscription is not restored*

## Actual result 

If you have both subscription `Leo` and `VPN` available, and `Leo` credentials are restored first, then `VPN` credentials are not restored/not working even if it shows a proper message `You have active credentials loaded!`.

*Restore `Leo` then `VPN` subscription*

https://github.com/brave/brave-browser/assets/128532965/f1b3b1a2-ea4b-4607-911a-ea2c7b57b356

https://github.com/brave/brave-browser/assets/128532965/146901aa-2a96-4e11-a78b-fe70c53c8b1b

## Expected result

Both subscription credentials `Leo` and `VPN` must be successfully restored.

*Restore `VPN` then `Leo` subscription*

https://github.com/brave/brave-browser/assets/128532965/7d346189-286a-4456-97e0-dad4537bc561

https://github.com/brave/brave-browser/assets/128532965/e7c8404a-3f69-49d0-9e66-f9de455c9425

## Issue reproduces how often 

`Easily reproduced`

## Version/Channel Information:

- Can you reproduce this issue with the current Play Store version? 
- Can you reproduce this issue with the current Play Store Beta version? 
- Can you reproduce this issue with the current Play Store Nightly version?

## Device details
   - Install type (ARM, x86): ARM
   - Device type (Phone, Tablet, Phablet): Pixel 7 (phone)
   - Android version: Android 14
   
## Brave version

```
Brave build: 1.64.113
Chromium: 123.0.6312.86 (Official Build) (64-bit) 
```

### Website problems only
- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards?
- Is the issue reproducible on the latest version of Chrome? 

### Additional information

cc @deeppandya 

I reached the limit and can't check network logs via chrome://inspect today. I'll try to do it when I unblock.

**Comment by hffvld** [comment]
Hi @deeppandya no suspicious calls in Dev tools

![2024-03-29_13-18-55](https://github.com/brave/brave-browser/assets/128532965/ffd52063-ba0f-47e5-8f26-c2b81de7259f)

Here is HAR file just in case [Leo and VPN subscription.har.zip](https://github.com/brave/brave-browser/files/14808295/Leo.and.VPN.subscription.har.zip)

**Comment by mattmcalister** [comment]
I believe this is no longer relevant...can you confirm @hffvld ?

**Comment by hffvld** [comment]
Hey @mattmcalister 

I checked 2 different scenarios:
- Purchase `VPN` and `Leo` subscriptions on the `Web` > Login to web account >  Tap the `Refresh` button for `Leo` then for `VPN` > Verified that both subscriptions are restored and works as expected.
- Purchase `VPN` and `Leo` subscriptions on the `Web` > Open `Leo` settings > Tap `Refresh credentials` > Open `VPN` settings > Tap `Refresh credentials` > Verified that both subscriptions are restored and works as expected.

and I wasn't able to reproduce the issue.

-------------------------------------
```
Device/OS: Pixel 7 / panther_beta-user 15 AP41.240823.009 release-keys
Brave build: 1.71.113
Chromium: 130.0.6723.58 (Official Build) (64-bit) 
```
------------------------------------

https://github.com/user-attachments/assets/e6b79a90-7aa2-42e4-927f-859b43957765

https://github.com/user-attachments/assets/14a3c74b-1d34-4210-bc45-6dd28ff08536
